File: apps/Applications/ios/Local For Vocal/Local For Vocal/Sources/Views/Components/GlowForHarvestView.swift
import SwiftUI

struct GlowForHarvestView: View {
    struct HarvestItem: Decodable, Identifiable {
        let id: String
        let name: String
        let image: String
        let offer: String
        let actionUrl: String?
    }

    let title: String
    let headerActionUrl: String?
    let items: [HarvestItem]

    var body: some View {
        VStack(alignment: .leading, spacing: 12) {
            // Header
            HStack {
                Text(title)
                    .font(.system(size: 18, weight: .bold))
                    .foregroundColor(Color(hex: "#000000"))

                Spacer()

                if let action = headerActionUrl {
                    Button(action: {
                        AppLogger.debug("Navigate to: \(action)")
                    }) {
                        Text("View All")
                            .font(.system(size: 14, weight: .semibold))
                            .foregroundColor(Color(hex: "#FF6F00"))
                    }
                }
            }
            .padding(.horizontal, 16)

            // Scroll Content
            ScrollView(.horizontal, showsIndicators: false) {
                HStack(spacing: 12) {
                    ForEach(items) { item in
                        HarvestCard(item: item)
                    }
                }
                .padding(.horizontal, 16)
            }
        }
        .padding(.bottom, 20)
    }
}

struct HarvestCard: View {
    let item: GlowForHarvestView.HarvestItem

    var body: some View {
        Button(action: {
            if let action = item.actionUrl {
                AppLogger.debug("Navigate to: \(action)")
            }
        }) {
            VStack(spacing: 8) {
                // Card
                ZStack {
                    // 1. Background Image Pattern
                    AsyncImage(
                        url: URL(
                            string:
                                "https://res.cloudinary.com/deljcbcvu/image/upload/v1768414560/IMG_1856_tkqhto.jpg"
                        )
                    ) { image in
                        image.resizable()
                            .aspectRatio(contentMode: .fill)
                    } placeholder: {
                        Color(hex: "#FFF3E0")
                    }
                    .opacity(0.6)

                    // 2. Kites Decoration
                    VStack {
                        HStack {
                            Spacer()
                            Text("🪁")
                                .font(.system(size: 24))
                                .rotationEffect(.degrees(15))
                                .offset(x: 10, y: -10)
                        }
                        Spacer()
                    }
                    VStack {
                        HStack {
                            Text("🪁")
                                .font(.system(size: 16))
                                .rotationEffect(.degrees(-10))
                                .offset(x: -10, y: 10)
                            Spacer()
                        }
                    }

                    // 3. Main Product Image
                    AsyncImage(url: URL(string: item.image)) { image in
                        image.resizable()
                            .aspectRatio(contentMode: .fit)
                    } placeholder: {
                        Color.clear
                    }
                    .frame(width: 90, height: 90)

                    // 4. Offer Badge
                    VStack {
                        Spacer()
                        Text(item.offer)
                            .font(.system(size: 12, weight: .bold))
                            .foregroundColor(.white)
                            .frame(maxWidth: .infinity)
                            .padding(.vertical, 6)
                            .background(
                                Ellipse()
                                    .fill(Color(hex: "#E91E63"))
                                    .frame(height: 40)
                                    .offset(y: 10)
                            )
                    }
                }
                .frame(width: 140, height: 180)
                .background(Color(hex: "#FFF3E0"))
                .cornerRadius(16)
                .clipped()

                // Name Pill
                Text(item.name)
                    .font(.system(size: 14, weight: .medium))
                    .foregroundColor(Color(hex: "#333333"))
                    .padding(.horizontal, 16)
                    .padding(.vertical, 4)
                    .background(Color(hex: "#FCE4EC"))
                    .cornerRadius(12)
            }
        }
        .buttonStyle(PlainButtonStyle())
    }
}

File: apps/Applications/android/app/src/main/java/com/ecommerceearn/app/ui/components/GlowForHarvestView.kt
package com.ecommerceearn.app.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyRow
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.alpha
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.rotate
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import coil.compose.AsyncImage

data class GlowForHarvestItem(
    val id: String,
    val name: String,
    val image: String,
    val offer: String,
    val actionUrl: String?
)

@Composable
fun GlowForHarvestView(
    title: String,
    headerActionUrl: String?,
    items: List<GlowForHarvestItem>
) {
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .padding(bottom = 20.dp),
        verticalArrangement = Arrangement.spacedBy(12.dp)
    ) {
        // Header
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(horizontal = 16.dp),
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.SpaceBetween
        ) {
            Text(
                text = title,
                fontSize = 18.sp,
                fontWeight = FontWeight.Bold,
                color = Color(0xFF000000)
            )

            if (headerActionUrl != null) {
                Text(
                    text = "View All",
                    fontSize = 14.sp,
                    fontWeight = FontWeight.SemiBold,
                    color = Color(0xFFFF6F00),
                    modifier = Modifier.clickable { 
                        // Handle navigation to headerActionUrl
                    }
                )
            }
        }

        // Scroll Content
        LazyRow(
            contentPadding = PaddingValues(horizontal = 16.dp),
            horizontalArrangement = Arrangement.spacedBy(12.dp)
        ) {
            items(items) { item ->
                GlowForHarvestCard(item = item)
            }
        }
    }
}

@Composable
fun GlowForHarvestCard(item: GlowForHarvestItem) {
    Column(
        modifier = Modifier.clickable {
            item.actionUrl?.let {
                // Navigate
            }
        },
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.spacedBy(8.dp)
    ) {
        // Card ZStack Equivalent
        Box(
            modifier = Modifier
                .size(140.dp, 180.dp)
                .clip(RoundedCornerShape(16.dp))
                .background(Color(0xFFFFF3E0))
        ) {
            // 1. Background Image Pattern
            AsyncImage(
                model = "https://res.cloudinary.com/deljcbcvu/image/upload/v1768414560/IMG_1856_tkqhto.jpg",
                contentDescription = null,
                contentScale = ContentScale.Crop,
                modifier = Modifier
                    .fillMaxSize()
                    .alpha(0.6f)
            )

            // 2. Kites Decoration
            Box(modifier = Modifier.fillMaxSize()) {
                Text(
                    text = "🪁",
                    fontSize = 24.sp,
                    modifier = Modifier
                        .align(Alignment.TopEnd)
                        .offset(x = 10.dp, y = (-10).dp)
                        .rotate(15f)
                )

                Text(
                    text = "🪁",
                    fontSize = 16.sp,
                    modifier = Modifier
                        .align(Alignment.BottomStart)
                        .offset(x = (-10).dp, y = 10.dp)
                        .rotate(-10f)
                )
            }

            // 3. Main Product Image
            AsyncImage(
                model = item.image,
                contentDescription = item.name,
                contentScale = ContentScale.Fit,
                modifier = Modifier
                    .size(140.dp)
                    .align(Alignment.Center)
            )

            // 4. Offer Badge
            Box(
                modifier = Modifier
                    .align(Alignment.BottomCenter)
                    .offset(y = 10.dp)
                    .fillMaxWidth(0.9f)
                    .height(40.dp)
                    .background(Color(0xFFE91E63), RoundedCornerShape(50)),
                contentAlignment = Alignment.Center
            ) {
                Text(
                    text = item.offer,
                    fontSize = 12.sp,
                    fontWeight = FontWeight.Bold,
                    color = Color.White,
                    modifier = Modifier.padding(vertical = 6.dp)
                )
            }
        }

        // Name Pill
        Text(
            text = item.name,
            fontSize = 14.sp,
            fontWeight = FontWeight.Medium,
            color = Color(0xFF333333),
            modifier = Modifier
                .background(Color(0xFFFCE4EC), RoundedCornerShape(12.dp))
                .padding(horizontal = 16.dp, vertical = 4.dp)
        )
    }
}
